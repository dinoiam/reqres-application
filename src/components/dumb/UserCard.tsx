import React, { memo } from 'react';
import styled from 'styled-components';

const UserCardComponent = styled.div`
  border: 1px solid black;
  border-radius: 5px;
  display: flex;
  max-width: 400px;
  min-height: 100px;
  cursor: pointer;
`;

const UserCardInfo = styled.div`
  width: 300px;
  padding: 10px;
  font-size: 14px;
`;

type Props = {
  /** Image url to show inside the card */
  image: string;
  /** A string representing the fist name */
  firstName: string;
  /** A string representing the last name */
  lastName: string;
  /** A string representing the email name */
  email: string;
  /** Callback function called on click on the card */
  onClick: () => void;
};

export const UserCard = memo(
  function UserCard({ firstName, image, lastName, email, onClick }: Props): JSX.Element {
    return (
      <UserCardComponent onClick={onClick}>
        <img src={image}></img>
        <UserCardInfo>
          <h3>
            {firstName} {lastName}
          </h3>
          <h4>{email}</h4>
        </UserCardInfo>
      </UserCardComponent>
    );
  },
  () => true
);
