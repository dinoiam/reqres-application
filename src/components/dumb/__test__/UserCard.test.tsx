import React from 'react';
import { render } from '@testing-library/react';
import { UserCard } from '../UserCard';

describe('UserCard', () => {
  test('snapshot', () => {
    const props = {
      image: 'test_image',
      firstName: 'test_firstName',
      lastName: 'test_lastName',
      email: 'test_email',
      onClick: jest.fn()
    };
    const { container } = render(<UserCard {...props}></UserCard>);

    expect(container).toMatchSnapshot();
  });
});
