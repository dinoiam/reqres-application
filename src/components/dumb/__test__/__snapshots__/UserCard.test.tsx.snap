// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard snapshot 1`] = `
<div>
  <div
    class="sc-bdfBQB gDgFmX"
  >
    <img
      src="test_image"
    />
    <div
      class="sc-gsTEea kymFbO"
    >
      <h3>
        test_firstName
         
        test_lastName
      </h3>
      <h4>
        test_email
      </h4>
    </div>
  </div>
</div>
`;
